test/irmin-pack/migration.ml

```diff
@@ -0,0 +1,100 @@
+open Lwt.Infix
+open Common
+
+let src = Logs.Src.create "tests.migration" ~doc:"Test migrations"
+
+module Log = (val Logs.src_log src : Logs.LOG)
+
+module Hash = Irmin.Hash.SHA1
+module S =
+  Irmin_pack.Make (Conf) (Irmin.Metadata.None) (Irmin.Contents.String)
+    (Irmin.Path.String_list)
+    (Irmin.Branch.String)
+    (Hash)
+
+let config ?(readonly = false) ?(fresh = true) root =
+  Irmin_pack.config ~readonly ?index_log_size ~fresh root
+
+let info () = Irmin.Info.empty
+
+let root_V1_source, root_V1, root_V2 =
+  let ( / ) = Filename.concat in
+  ("data" / "version_1", "_build" / "version_1", "_build" / "version_2")
+
+(** testV1 is a store in V1 used to test the migration to V2 *)
+let setup_test_env () =
+  rm_dir root_V1;
+  rm_dir root_V2;
+  let cmd = Printf.sprintf "cp -rp %s %s" root_V1_source root_V1 in
+  Fmt.epr "exec: %s\n%!" cmd;
```

Comment:
Need to quote such commands with `Filename.quote_command` for this to work with certain filepaths on Windows.

Reply:
It seems that this is only included in [ocaml 4.10](https://discuss.ocaml.org/t/ocaml-4-10-released/5194)